test/parallel/test-child-process-spawn-name.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+// This test spawns itself with an argument to indicate when it is a child to
+// easily and portably print the process title
+if (process.argv.length === 3) {
+  console.log(process.title);
+  process.exit();
+}
+
+const noCustomName = cp.spawn(process.execPath,
+    ['test/parallel/test-child-process-spawn-name.js', 'child']);
+
+let noCustomNameOutput = '';
+noCustomName.stdout.on('data', (data) => {
+  noCustomNameOutput += data;
+});
+noCustomName.on('close', common.mustCall((code, signal) => {
+  assert.strictEqual(noCustomNameOutput.trim(), process.execPath);
+}));
+
+
+const customName = cp.spawn(process.execPath,
+    ['test/parallel/test-child-process-spawn-name.js', 'child'],
+    {name: 'customName'});
+
+let customNameOutput = '';
+customName.stdout.on('data', (data) => {
+  customNameOutput += data;
+});
+customName.on('close', common.mustCall((code, signal) => {
+  assert.strictEqual(customNameOutput.trim(), 'customName');
```

Comment:
I'd be curious to see if this works on SmartOS. IIRC, SmartOS does not support changing `process.title`